types/react/index.d.ts

```diff
@@ -418,6 +418,31 @@ declare namespace React {
         refs: {
             [key: string]: ReactInstance
         };
+
+        // Static properties copied from ComponentClass interface.
+        /**
+         * The validity of the type of `propTypes` cannot be checked
+         * automatically at the definition of a component class because the
+         * required type depends on `P`.  We recommend checking it using
+         * `asPropTypes` from the `react-typescript-helpers` package.  If you
+         * don't do that, then the type will still be checked by
+         * JSX.LibraryManagedAttributes every time the component is used,
+         * assuming you use TypeScript 3.0 or newer.
+         */
+        static propTypes?: any;
```

Comment:
Would `object` be better?

Reply:
`object` broke the tests because a component class that did not redeclare `propTypes` or `defaultProps` was not assignable to `ComponentClass<P>`.  I'm changing this back to `any`.